src/components/interfaces/HMI_API.xml

```diff
@@ -3281,6 +3300,22 @@
       <description>ID of application related to this RPC.</description>
     </param>
   </function>
+  <function name="SendHapticData" messagetype="request">
+    <description>Send the UI spatial data from MOBILE. This data will be utilized by the HMI to determine how and when haptic events should occur</description>
+    <param name="appID" type="Integer" mandatory="true">
+      <description>Internal ID of the application that requested this RPC.</description>
+    </param>
+    <param name="HapticSpatialData" type="Common.SpatialStruct" minsize="0" maxsize="1000" mandatory="false" array="true">
+      <description>
+        Array of spatial data structures that represent the locations of all user controls present on the HMI.
+        This data should be updated if/when the application presents a new screen.
+        When a request is sent, if successful, it will replace all spatial data previously sent through RPC.
+        Avoidance of doubt, when an empty HapticSpatialData, it will be clear all spatial data previously sent through RPC.
+      </description>
+    </param>
+  </function>
+  <function name="SendHapticData" messagetype="response">
+  </function>
 </interface>
 
 <interface name="Navigation" version="1.4.0" date="2017-04-27"> 
```

Comment:
Also, this will require you to rebase, because the overarching struct `VideoStreamingCapability` is not defined in your tree